common/src/metrics_helpers.rs

```diff
@@ -0,0 +1,82 @@
+//! Helper functions and constants for recording metrics throughout the relayer
+
+use circuit_types::{r#match::MatchResult, transfers::ExternalTransferDirection};
+use num_bigint::BigUint;
+use util::hex::biguint_to_hex_string;
+
+use crate::types::{token::Token, transfer_auth::ExternalTransferWithAuth};
+
+/// Metric describing the number of deposits made
+const NUM_DEPOSITS_METRICS: &str = "num_deposits";
+/// Metric describing the volume of deposits made
+const DEPOSIT_VOLUME_METRIC: &str = "deposit_volume";
+/// Metric describing the number of withdrawals made
+const NUM_WITHDRAWALS_METRICS: &str = "num_withdrawals";
+/// Metric describing the volume of withdrawals made
+const WITHDRAWAL_VOLUME_METRIC: &str = "withdrawal_volume";
+/// Metric describing the volume of the base asset in a match
+const MATCH_BASE_VOLUME_METRIC: &str = "match_base_volume";
+/// Metric describing the volume of the quote asset in a match
+const MATCH_QUOTE_VOLUME_METRIC: &str = "match_quote_volume";
+/// Metric label for the asset of a deposit/withdrawal
+const ASSET_METRIC_LABEL: &str = "asset";
+
+/// Get the human-readable asset and volume of
+/// the given mint and amount.
+/// The asset is the token ticker, if it is found, otherwise
+/// the token's address.
+/// The amount is the decimal amount of the transfer, going through
+/// lossy f64 conversion via the associated number of decimals
+fn get_asset_and_volume(mint: &BigUint, amount: u128) -> Result<(String, f64), String> {
+    let token = Token::from_addr_biguint(mint);
+    let asset = token.get_ticker().unwrap_or(&biguint_to_hex_string(mint)).to_string();
+    let volume = token.get_amount_f64(amount)?;
+
+    Ok((asset, volume))
+}
+
+/// Record the volume of an asset movement (e.g. deposit, withdrawal, trade)
```

Comment:
nit: wording here feels a bit weird

```suggestion
/// Record a volume metric (e.g. deposit, withdrawal, trade)
```

Reply:
Also we can make the method it calls infallible via the suggestion above, so we can rename this method